Add PodSecurityPolicy for crd manager jobs

bashofmann opened the pull request:
Add PodSecurityPolicy for crd manager jobs so that it can be installed into a hardened cluster with PodSecurityPolicies.

Addresses https://github.com/rancher/rancher/issues/29980 and https://github.com/rancher/rancher/issues/30063

bashofmann commented:
@aiyengar2 @prachidamle I took a stab at creating a PodSecurityPolicy for the crd manager jobs to address rancher/rancher#29980. If you have the time, it would be great if you could have a look and review it.

aiyengar2 reviewed (commented):
This looks great! Thank you for the contribution @bashofmann.

The CRD template changes itself are sound from what I can tell, but I would need to pull it in and run the scripts to test out if the CRD chart deployment works as expected. e.g.
- `make CHART=rancher-monitoring charts`
- Check if `helm template packages/rancher-monitoring/charts-crd` works
- Push changes to your branch
- Add the branch as a Helm Repo on the Dashboard UI on some hardened cluster
- Try to deploy the version of Monitoring from your branch following the details in https://github.com/rancher/rancher/issues/29980 and see if the CRD chart is successfully deployed along with the rest of monitoring

Also, we'll probably need to bump the `packageVersion` at `packages/rancher-monitoring/package.yaml` since the current `packageVersion` of `01` is [already released in main](https://github.com/rancher/charts/tree/main/assets/rancher-monitoring).

This will allow QA to test the changes as well.

Related Issue: https://github.com/rancher/rancher/issues/29980

bashofmann commented:
I tested the PSP manually before to ensure it's working. Thanks @aiyengar2 for describing how to correctly compile and test it :-):

* I bumped the packageVersion
* Compiled the charts and pushed them to a new branch https://github.com/bashofmann/charts-1/tree/crd-manager-psp-compiled2
* Added the branch as a repo to a cluster with PSP activated
* Install monitoring from there

The PSP for the crd manager job works, but then I noticed that the alertmanager-rancher-monitoring-pre-install Job is also missing a PSP, so I added one as well. After adding this, installation works successfully.

bashofmann commented:
I was too quick. The prometheus-adapater deployment is also missing a PSP.

bashofmann commented:
I found more missing PSPs. I now added PSPs for:
* prometheus-adapter
* pushproxy proxies
* rancher-monitoring-patch-sa Job

Now the helm installation went through successfully and all deployments, daemonsets and statefulsets are running.

Branch for testing: https://github.com/bashofmann/charts-1/tree/crd-manager-psp-compiled8

In order to add a PSP for prometheus adapter, I updated the chart to 2.7.1 and changed it to the new location in https://github.com/prometheus-community/helm-charts (https://github.com/helm/charts/tree/master/stable/prometheus-adapter is deprecated).
The new chart version also comes with a new version of the image (v0.7.0). This new image version is not present yet in our rancher org on docker hub. I tested with v0.6.0 that the PSP works.

bashofmann commented:
Pull request for rancher/image-mirror: https://github.com/rancher/image-mirror/pull/51

aiyengar2 commented on packages/rancher-monitoring/rancher-monitoring.patch:
```diff
@@ -512,8 +491,8 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/req
      condition: grafana.enabled
 +
 +  - name: prometheus-adapter
```
`rancher-pushprox` also needs a `version` bump on the `Chart.yaml` as well as version bumps throughout the dependencies file since it is also maintained in `rancher/charts` as a separate chart. (Note: It probably didn't affect you when testing due to the way it is packaged via local chart archives)

Also, @bashofmann I was planning to make a separate PR updating the repositories in https://github.com/rancher/charts/pull/846 but would it be possible for you to just update this PR with the changes? `rancher/charts` doesn't work nicely with multiple people making PRs to update the same file in a patch, so that would be easier than performing a rebase after one or the other is merged in.

bashofmann replied:
I bumped the version of `ranher-pushprox` in all places I could find and cherry-picked your commit from #846 into this branch.

aiyengar2 reviewed (commented):
Wow, this is awesome! Thank you for looking into this @bashofmann.

Overall, this PR looks good to me. I had two minor comments regarding versions and a missing newline but once those are addressed it should be good to merge since it has been tested as well.

aiyengar2 reviewed (approved):
LGTM! Closing out https://github.com/rancher/charts/pull/846 since the changes have already been merged here.

This PR also now resolves the deprecated Helm repos in Monitoring V2.

Related Issue: https://github.com/rancher/rancher/issues/30063

aiyengar2 commented:
Merging this PR in once https://github.com/rancher/image-mirror/pull/51 is merged.
